jdbc/src/main/scala/zio/sql/TransactionModule.scala

```diff
@@ -87,15 +87,28 @@ trait TransactionModule { self: Jdbc =>
         ZTransaction.fromEffect(coreDriver.updateOn(update, connection))
       }
 
+    def apply(update: List[self.Update[_]]): ZTransaction[Any, Exception, List[Int]] =
+      txn.flatMap { case Txn(connection, coreDriver) =>
+        ZTransaction.fromEffect(coreDriver.updateOnBatch(update, connection))
+      }
     def apply[Z: Schema](insert: self.Insert[_, Z]): ZTransaction[Any, Exception, Int] =
       txn.flatMap { case Txn(connection, coreDriver) =>
         ZTransaction.fromEffect(coreDriver.insertOn(insert, connection))
       }
 
+    def apply[Z: Schema](insert: List[self.Insert[_, Z]]): ZTransaction[Any, Exception, List[Int]] =
+      txn.flatMap { case Txn(connection, coreDriver) =>
+        ZTransaction.fromEffect(coreDriver.insertOnBatch(insert, connection))
+      }
+
     def apply(delete: self.Delete[_]): ZTransaction[Any, Exception, Int] =
       txn.flatMap { case Txn(connection, coreDriver) =>
         ZTransaction.fromEffect(coreDriver.deleteOn(delete, connection))
       }
+    def applyBatch(delete: List[self.Delete[_]]): ZTransaction[Any, Exception, List[Int]] =
```

Comment:
I would just call it apply, not applyBatch.

Reply:
ok

Reply:
the issue here is  i cannot use the same apply method for update and delete
as these method names would collide after type erasure
```
 double definition:
[error] def applyBatch(update: List[TransactionModule.this.Update[_]]): TransactionModule.this.ZTransaction[Any,Exception,List[Int]] at line 90 an
[error] def applyBatch(delete: List[TransactionModule.this.Delete[_]]): TransactionModule.this.ZTransaction[Any,Exception,List[Int]] at line 108
[error] have same type after erasure: (update: List): zio.sql.TransactionModule#ZTransaction
[error]     def applyBatch(delete: List[self.Delete[_]]): ZTransaction[Any, Exception, List[Int]] =
```

Reply:
I see but the names are really sub optimal. From the user perspective, its hard to discover these methods.

I cannot think of much better names but probably these are better:
- ZTransaction.batchUpdate
- ZTransaction.batchDelete
- ZTransaction.batchInsert

What do you think?

Reply:
fine.
Let me rename the apply method accordingly